src/main/java/jenkins/plugins/publish_over_cifs/CifsPublisherPlugin.java

```diff
@@ -91,14 +120,20 @@ public String toString() {
     }
 
     public Descriptor getDescriptor() {
-        return Hudson.getInstance().getDescriptorByType(Descriptor.class);
+        Jenkins jenkins = Jenkins.getInstance();
```

Comment:
This seems to be repeated, maybe make it into a method?

Reply:
Is your intention to replace the getDescriptor and alike methods with one static helper method in all classes in the plugin?

Reply:
Yes, you could make it a static method of the publisher class or the descriptor class. Either one would work.

Reply:
I created a separate class to host the common nullsafe decriptor fetching logic. Otherwise I had introduced dependencies between classes that had no dependencies before.